versioned/tiered/tiered-impl/src/test/java/org/projectnessie/versioned/impl/condition/TestUpdateClauseVisitor.java

```diff
@@ -0,0 +1,153 @@
+      // An operator to set some part or all of an entity.
+      SET
+    }
+
+    Operator getOperator();
+
+    ExpressionPath getPath();
+
+    class SetCommand implements UpdateCommand {
+      private final Entity entity;
+      private final ExpressionPath path;
+
+      SetCommand(ExpressionPath path, Entity entity) {
+        this.path = path;
+        this.entity = entity;
+      }
+
+      @Override
+      public Operator getOperator() {
+        return Operator.SET;
+      }
+
+      @Override
+      public ExpressionPath getPath() {
+        return path;
+      }
+
+      Entity getEntity() {
+        return entity;
+      }
+    }
+
+    class RemoveCommand implements UpdateCommand {
+      private final ExpressionPath path;
+
+      RemoveCommand(ExpressionPath path) {
+        this.path = path;
+      }
+
+      @Override
+      public Operator getOperator() {
+        return Operator.REMOVE;
+      }
+
+      @Override
+      public ExpressionPath getPath() {
+        return path;
+      }
+    }
+  }
+
+  /**
+   * A test visitor which builds up an UpdateCommand representation of the passed in UpdateClause.
+   */
+  static class Visitor implements UpdateClauseVisitor<UpdateCommand> {
+
+    @Override
+    public UpdateCommand visit(final RemoveClause clause) {
+      return new UpdateCommand.RemoveCommand(clause.getPath());
+    }
+
+    @Override
+    public UpdateCommand visit(final SetClause clause) {
+      switch (clause.getValue().getType()) {
+        case VALUE:
+          return new UpdateCommand.SetCommand(clause.getPath(), clause.getValue().getValue());
+        case FUNCTION:
+          return new UpdateCommand.SetCommand(clause.getPath(), handleFunction(clause.getValue().getFunction()));
+        default:
+          throw new UnsupportedOperationException(String.format("Unsupported SetClause type: %s", clause.getValue().getType().name()));
+      }
+    }
+
+    private Entity handleFunction(ExpressionFunction expressionFunction) {
+      if (ExpressionFunction.FunctionName.LIST_APPEND == expressionFunction.getName()) {
+        return expressionFunction.getArguments().get(1).getValue();
+      }
+      throw new UnsupportedOperationException(String.format("Unsupported Set function: %s", expressionFunction.getName()));
+    }
+  }
+
+  @Test
+  void testRemoveClause() {
+    final UpdateClause clause = RemoveClause.of(PATH);
+    final UpdateCommand command = clause.accept(VISITOR);
+    Assertions.assertEquals(UpdateCommand.Operator.REMOVE, command.getOperator());
+    Assertions.assertEquals(PATH, command.getPath());
+  }
+
+  @Test
+  void testSetClauseEquals() {
+    testSetClause(SetClause.equals(PATH, ENTITY_STR), PATH, ENTITY_STR);
+  }
+
+  @Test
+  void testSetClauseAppendToList() {
+    testSetClause(SetClause.appendToList(PATH, ENTITY_STR), PATH, ENTITY_STR);
+  }
+
+  void testSetClause(UpdateClause clause, ExpressionPath path, Entity entity) {
+    final UpdateCommand command = clause.accept(VISITOR);
```

Comment:
How come this isn't used for the RemoveClause as well?

Reply:
If you mean why isn't `testSetClause()` used for remove? Remove doesn't use an entity. Unless you are suggesting I use Optional instead?

Reply:
It's a minor nit, but it seems that you could remove the entity assertion and add it to the specific tests, or have another method call this one. This comment is minor, do what you think is best.

Reply:
I decided moving the assertion to the specific test was the cleanest.